lib/child_process.js

```diff
@@ -114,6 +109,82 @@ exports.exec = function(command /*, options, callback*/) {
                           opts.callback);
 };
 
+function normalizeArgsOptions(allArgs, defaultOption) {
```

Comment:
This seems like it is going to be a nightmare to maintain. If we can't do the validation more simply, I think we'd be better off skipping it or loosening its strictness.